dpytools/logger/logger.py

```diff
@@ -1,2 +1,183 @@
+# #4 Still need to work out how to format JSON logs as JSON Lines (may need to create a callable to pass to structlog.processors.JSONRenderer())
+# #4 TODO Add namespace to output - using an env var?
+def configure_logger(enable_console_logs: bool = HUMAN_FRIENDLY_LOGS):
+    shared_processors = [
+        structlog.processors.TimeStamper(fmt="iso", key="created_at"),
+        structlog.processors.UnicodeEncoder(),
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.add_logger_name,
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.CallsiteParameterAdder(
+            {
+                structlog.processors.CallsiteParameter.PATHNAME,
+                structlog.processors.CallsiteParameter.FILENAME,
+                structlog.processors.CallsiteParameter.MODULE,
+                structlog.processors.CallsiteParameter.FUNC_NAME,
+                structlog.processors.CallsiteParameter.THREAD,
+                structlog.processors.CallsiteParameter.THREAD_NAME,
+                structlog.processors.CallsiteParameter.PROCESS,
+                structlog.processors.CallsiteParameter.PROCESS_NAME,
+            }
+        ),
+        structlog.stdlib.ExtraAdder(),
+    ]
+
+    structlog.configure(
+        processors=shared_processors
+        + [structlog.stdlib.ProcessorFormatter.wrap_for_formatter],
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        wrapper_class=structlog.stdlib.BoundLogger,
+        cache_logger_on_first_use=True,
+    )
+    # TODO Pass a callable to JSONRenderer to format logs as JSON Lines
+    logs_render = (
+        structlog.dev.ConsoleRenderer(colors=True)
+        if enable_console_logs
+        else structlog.processors.JSONRenderer()
+    )
+
+    _configure_default_logging_by_custom(shared_processors, logs_render)
+
+
+def _configure_default_logging_by_custom(shared_processors, logs_render):
+    handler = logging.StreamHandler()
+    formatter = structlog.stdlib.ProcessorFormatter(
+        foreign_pre_chain=shared_processors,
+        processors=[
+            _extract_from_record,
+            structlog.stdlib.ProcessorFormatter.remove_processors_meta,
+            logs_render,
+        ],
+    )
+    handler.setFormatter(formatter)
+    root_uvicorn_logger = logging.getLogger()
+    root_uvicorn_logger.addHandler(handler)
+    root_uvicorn_logger.setLevel(logging.INFO)
+
+
+def _extract_from_record(_, __, event_dict):
+    record = event_dict["_record"]
+    event_dict["thread_name"] = record.threadName
+    event_dict["process_name"] = record.processName
+    return event_dict
+
+
+# #4 TODO Add namespace to create_log_event function arguments?
+# #4 Do we need to specify a format for, e.g., http_data? The DP logging standards include a list of fields for HTTP events (https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md#http-event-data), as well as auth events and errors
```

Comment:
a good question.

so errors we can do. So:

take error information in as part of logging.error() (there's a comment elsewhere, if you need more that just the exception adapt the signature as needed). Pass what you get to a `create_error_struct()` function to do the munging please.

for auth and http we cant do them as we dont have http and auth clients (so dont know what the inputs will be). So for those two add kwargs and functions in the same style -but- the functions can just return whatever dict they're given (just add a todo). We'll do structuring the auth and http sections of logs as a later task.